pkg/typesystem/typesystem.go

```diff
@@ -108,45 +81,75 @@ func (t *TypeSystem) GetRelation(objectType, relation string) (*openfgapb.Relati
 	return r, nil
 }
 
-// Validate validates the type system according to the following rules:
+// Validate validates an *openfgapb.AuthorizationModel and returns it, populating SchemaVersion if empty.
+//
+// The following rules are validated:
 //  1. For every rewrite the relations in the rewrite must:
-//     a. Be valid relations on the same type in the authorization typeSystem (in cases of computedUserset)
+//     a. Be valid relations on the same type in the authorization model (in cases of computedUserset)
 //     b. Be valid relations on another existing type (in cases of tupleToUserset)
-//  2. Do not allow duplicate types or duplicate relations (but that is inherent in the map structure so nothing to
-//     actually check)
+//  2. Do not allow duplicate types or duplicate relations (only need to check types as relations are
+//     in a map so cannot contain duplicates)
 //
-// If it is a SchemaVersion1_1 type system (with types on relations), then additionally validate the type system
+// If it is a 1.1 authorization model (with types on relations), then additionally validate the type system
 // according to the following rules:
 //  3. Every type on a relation must be a valid type:
 //     a. For a type (e.g. user) this means checking that this type is in the TypeSystem
 //     b. For a type#relation this means checking that this type with this relation is in the TypeSystem
 //  4. Check that a relation is assignable if and only if it has a non-zero list of types
-func (t *TypeSystem) Validate() error {
-	if err := t.validateRelationRewrites(); err != nil {
-		return err
+func Validate(model *openfgapb.AuthorizationModel) (*openfgapb.AuthorizationModel, error) {
+	switch model.SchemaVersion {
+	case "", "1.0":
+		model.SchemaVersion = "1.0"
+	case "1.1":
+	default:
+		return nil, ErrInvalidSchemaVersion
 	}
 
-	if t.Version == SchemaVersion1_1 {
-		if err := t.validateRelationTypeRestrictions(); err != nil {
-			return err
+	if containsDuplicateType(model) {
+		return nil, ErrDuplicateTypes
+	}
```

Comment:
```suggestion
	if err := containsDuplicateType(model); err != nil {
		return nil, err
	}
```

Reply:
`containsDuplicateType` returns a boolean

Reply:
I know, but to be consistent with the other validation methods, it could return an error